cmd/ls.go

```diff
@@ -0,0 +1,232 @@
+		"Show disk size in human readable format.",
+	)
+	f.StringSliceVar(&o.filters, "filter", []string{}, lsFilterFlagDesc)
+	f.Int64Var(&o.page, "page", 1, "Results page number (to be used with --size).")
+	f.Int64Var(&o.size, "size", 0, "Number of results per page (to be used with --page).")
+
+	return cmd
+}
+
+func (o *lsOptions) run(cmd *cobra.Command) error {
+	header := []string{"name", "size", "last-modified"}
+
+	filters, err := utils.NewFilters(nil, header, o.filters)
+	if err != nil {
+		return err
+	}
+	searchFilter, err := filters.GetJson(header)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Workflow %s selected", o.workflow)
+
+	lsParams := operations.NewGetFilesParams()
+	lsParams.SetAccessToken(&o.token)
+	lsParams.SetWorkflowIDOrName(o.workflow)
+	lsParams.SetFileName(&o.fileName)
+	lsParams.SetSearch(&searchFilter)
+	lsParams.SetPage(&o.page)
+	if cmd.Flags().Changed("size") {
+		lsParams.SetSize(&o.size)
+	}
+
+	api, err := client.ApiClient()
+	if err != nil {
+		return err
+	}
+	lsResp, err := api.Operations.GetFiles(lsParams)
+	if err != nil {
+		return err
+	}
+
+	parsedFormatFilters := utils.ParseFormatParameters(o.formatFilters)
+	if o.displayURLs {
+		displayLsURLs(cmd, lsResp.Payload, o.serverURL, o.workflow)
+	} else {
+		err = displayLsFiles(
+			cmd,
+			lsResp.Payload,
+			header,
+			parsedFormatFilters,
+			o.jsonOutput,
+			o.humanReadable,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func displayLsFiles(
+	cmd *cobra.Command,
+	p *operations.GetFilesOKBody,
+	header []string,
+	formatFilters []utils.FormatFilter,
+	jsonOutput bool,
+	humanReadable bool,
+) error {
+	var df dataframe.DataFrame
+	for _, col := range header {
+		colSeries := buildLsSeries(col, humanReadable)
+		for _, file := range p.Items {
+			if utils.HasAnyPrefix(file.Name, utils.FilesBlacklist) {
+				continue
+			}
+
+			var value any
+			switch col {
+			case "name":
+				value = file.Name
+			case "size":
+				if humanReadable {
+					value = file.Size.HumanReadable
+				} else {
+					value = int(file.Size.Raw)
+				}
+			case "last-modified":
+				value = file.LastModified
+			}
+
+			colSeries.Append(value)
+		}
+
+		df = df.CBind(dataframe.New(colSeries))
+	}
+
+	df = utils.FormatDataFrame(df, formatFilters)
+
+	if jsonOutput {
+		err := utils.DisplayJsonOutput(df.Maps(), cmd.OutOrStdout())
+		if err != nil {
+			return err
+		}
+	} else {
+		data := utils.DataFrameToStringData(df)
+		utils.DisplayTable(df.Names(), data, cmd.OutOrStdout())
+	}
+
+	return nil
+}
+
+func buildLsSeries(col string, humanReadable bool) series.Series {
+	if col == "size" && !humanReadable {
+		return series.New([]int{}, series.Int, col)
+	}
+	return series.New([]string{}, series.String, col)
+}
+
+func displayLsURLs(cmd *cobra.Command, p *operations.GetFilesOKBody, serverURL string, workflow string) {
+	for _, file := range p.Items {
+		fileURL := fmt.Sprintf("%s/api/workflows/%s/workspace/%s", serverURL, workflow, file.Name)
```

Comment:
Python client retrieves the URL from the swagger api but I didn't find a way to do it here, so I wrote it like this

Reply:
I don't think there is a way to do this right now. I also think that this is not needed in many places, so perhaps we can keep it like this and we can open an issue just for future reference?